playback/src/audio_backend/pulseaudio.rs

```diff
@@ -55,14 +52,27 @@ impl Sink for PulseAudioSink {
             return Ok(());
         }
 
+        let ss = pulse::sample::Spec {
+            format: self.pulse_format,
+            channels: NUM_CHANNELS,
+            rate: SAMPLE_RATE,
+        };
+
+        if !ss.is_valid() {
+            return Err(io::Error::new(
+                io::ErrorKind::Other,
+                "Invalid PulseAudio sample spec",
```

Comment:
In the Alsa backend you would have called this a `HwError` I think?

Reply:
I can do custom PulseAudio errors with `thiserror` in a different PR if you'd like? this PR was really about making sure it doesn't panic!.

Reply:
Why not do it in one swoop?

Reply:
OK. I'm fixing the clippy warnings 1st.

Reply:
> Why not do it in one swoop?

After you merge the clippy fixes and the pipe fixes (if the pipe fix PR is all good?) then I will put in some more work on the pulse backend and give it meaningful errors like I did with the alsa backend. I need to dig into the API(s) a bit so I know what the errors actually mean.

Reply:
@roderickvd all of the errors are returned directly to player I don't see the point of making custom errors? I read a little bit of the API doc and I made the messages as meaningful as I could.

Reply:
I went ahead and did it you win,lol!!! Errors are nice an pretty now with `thiserror`.